mailpoet/lib/Validator/Schema/ObjectSchema.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Validator\Schema;
+
+use MailPoet\Validator\Schema;
+
+// See: https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#objects
+class ObjectSchema extends Schema {
+  protected $schema = [
+    'type' => 'object',
+  ];
+
+  /** @param array<string, Schema> $properties */
+  public function properties(array $properties): self {
+    return $this->updateSchemaProperty('properties', array_map(
+      function (Schema $property) {
+        return $property->toArray();
+      },
+      $properties
+    ));
+  }
+
+  public function additionalProperties(Schema $schema): self {
+    return $this->updateSchemaProperty('additionalProperties', $schema->toArray());
+  }
+
+  public function enableAdditionalProperties(): self {
+    if (($this->schema['additionalProperties'] ?? []) === false) {
+      return $this->unsetSchemaProperty('additionalProperties');
+    }
+    return clone $this;
+  }
+
+  public function disableAdditionalProperties(): self {
+    return $this->updateSchemaProperty('additionalProperties', false);
+  }
+
+  /** @param array<string, Schema> $properties */
```

Comment:
That's a good point, I'll have a look and add more links.

> Some of the sections are in confusing places, like [format](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#format) only seems to apply to strings yet it's not in the Strings section.
This may originate from a bit of a strange nature of JSON schema itself. It's completely valid to specify `type` as `["string", "integer", "number", "boolean", ...]` and then add things like `format` or `minimum`, etc. Each of those modifiers will be applied only on the type that supports it though and for `format` it's indeed just for strings.

The schema builder I created is more restrictive and doesn't allow this type of mixing so it's clear what's being used and what's possible for each type. It shouldn't compromise on flexibility as we still have `oneOf`/`anyOf`, etc.

Reply:
Added info for object `patternProperties` as well as string `pattern`.

Reply:
Thanks for the clarification on the JSON schema and for adding extra info!